src/genomedata_load_data.c

```diff
@@ -117,6 +117,42 @@ typedef struct {
 #endif /* GCC_VERSION >= 20500*/
 #endif /* __GNUC__ */
 
+void load_data(char* gdfilename, char* trackname, bool verbose);
+
+static PyObject* load_data_from_stdin(PyObject* self, PyObject* args) {
+  const char* gdfilename;
+  const char* trackname = "dummy";
+  int verbose; /* needs to be int, not bool for parsing */
+
+  if (!PyArg_ParseTuple(args, "ssp", &gdfilename, &trackname, &verbose)) /* string string bool predicate */
+    return NULL;
+
+  load_data(gdfilename, trackname, (bool)verbose);
+
+  /* Return None in Python */
+  Py_RETURN_NONE;
+}
+
+static PyMethodDef LoadDataMethods[] = {
+    {"load_data_from_stdin", load_data_from_stdin, METH_VARARGS,
+     "Loads data in BED or WIG format from stdin into a track in a genomedata archive"},
+    {NULL, NULL, 0, NULL} /* Sentinel */
+};
+
+static struct PyModuleDef loadDateModule = {
+    PyModuleDef_HEAD_INIT,
+    "_load_data_c_ext", /* name of module */
+    NULL,               /* module documentation, may be NULL */
+    -1,                 /* size of per-interpreter state of the module,
+          or -1 if the module keeps state in global variables. */
+    LoadDataMethods };
+
+PyMODINIT_FUNC
+PyInit__load_data_c_ext(void) /* name is important for ref on import */
+{
+  return PyModule_Create(&loadDateModule);
+}
+
```

Comment:
Would be better to use the Python heap instead of `malloc()`

Reply:
I'm not certain what the specific benefits would be. It would allow the python process to garbage collect and/or do additional management if necessary. On the other hand it would increase some unknown amount of overhead.

If we decide to use the python allocator, all `free` calls need to be subsequently replaced as well with the Python API equivalent.

Reply:
OK, let's keep C `malloc()`/`free()`